libazureinit/src/provision/config.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
I am totally good w this but just to clarify, do you mean putting it next to main.rs in the primary azure-init/src folder? I could be totally wrong but I think if we do this, we have to have a lib.rs file that has both listed as modules or add mod config.rs inside of main.rs. 

I also think I may need to do a little research on how to import the config.rs mod into the libazureinit library. It's tricky because I think the library crate can't automatically use things from the main crate since libraries are supposed to be reusable without assuming a specific main binary.